Supporting create an RPC gang of size 1

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32731 Supporting create an RPC gang of size 1**
* #37184 Fix const-cast lint error in process_group_agent.cpp

As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

Differential Revision: [D19609558](https://our.internmc.facebook.com/intern/diff/D19609558)

mrshenli committed:
Supporting create an RPC gang of size 1

As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 0c35ba6e:


**None of the build failures appear to be your fault**.

* **1/1** recognized as flaky :snowflake:
    * Re-run these jobs?


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=0c35ba6eb2357693a098739c8804596c647e6845).

### :snowflake: 1 failure recognized as flaky
The following build failures have been detected as flaky and **may not be your fault**:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4363504) pytorch_linux_xenial_py3_6_gcc5_4_test (1/1)
**Step:** "Test" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=92224064) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=92224064)) :snowflake:
<details>
<summary>
<code>Jan 28 23:21:21 RuntimeError: quantization scale should be > 0</code>
</summary>

```
Jan 28 23:21:21   File "test_quantization.py", line 730, in test_quantized_rnn 
Jan 28 23:21:21     y, (h, c) = cell_dq(x, (h, c)) 
Jan 28 23:21:21   File "/opt/conda/lib/python3.6/site-packages/torch/nn/modules/module.py", line 540, in __call__ 
Jan 28 23:21:21     result = self.forward(*input, **kwargs) 
Jan 28 23:21:21   File "/opt/conda/lib/python3.6/site-packages/torch/nn/quantized/dynamic/modules/rnn.py", line 373, in forward 
Jan 28 23:21:21     return self.forward_tensor(input, hx) 
Jan 28 23:21:21   File "/opt/conda/lib/python3.6/site-packages/torch/nn/quantized/dynamic/modules/rnn.py", line 334, in forward_tensor 
Jan 28 23:21:21     input, hx, batch_sizes, max_batch_size, sorted_indices) 
Jan 28 23:21:21   File "/opt/conda/lib/python3.6/site-packages/torch/nn/quantized/dynamic/modules/rnn.py", line 315, in forward_impl 
Jan 28 23:21:21     self.batch_first, dtype=self.dtype, use_dynamic=True) 
Jan 28 23:21:21 RuntimeError: quantization scale should be > 0 
Jan 28 23:21:21  
Jan 28 23:21:21 ---------------------------------------------------------------------- 
Jan 28 23:21:21 Ran 36 tests in 59.421s 
Jan 28 23:21:21  
Jan 28 23:21:21 FAILED (errors=1, skipped=1) 
Jan 28 23:21:21  
Jan 28 23:21:21 Generating XML reports... 
Jan 28 23:21:21 Traceback (most recent call last): 
Jan 28 23:21:21   File "test/run_test.py", line 456, in <module> 
Jan 28 23:21:21     main() 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

mrshenli committed:
Update on "Supporting create an RPC gang of size 1"


As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

Differential Revision: [D19609558](https://our.internmc.facebook.com/intern/diff/D19609558)

[ghstack-poisoned]

zzzwen reviewed (approved):
LGTM

Commit:
Update on "Supporting create an RPC gang of size 1"


As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

Differential Revision: [D19609558](https://our.internmc.facebook.com/intern/diff/D19609558)

[ghstack-poisoned]

Commit:
Update on "Supporting create an RPC gang of size 1"


As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

Differential Revision: [D19609558](https://our.internmc.facebook.com/intern/diff/D19609558)

[ghstack-poisoned]

Commit:
Update on "Supporting create an RPC gang of size 1"


As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

Differential Revision: [D19609558](https://our.internmc.facebook.com/intern/diff/D19609558)

[ghstack-poisoned]

Commit:
Update on "Supporting create an RPC gang of size 1"


As we now support send to self, we no longer require world_size > 1.
Removing the assert from ProcessGroupAgent.

Differential Revision: [D19609558](https://our.internmc.facebook.com/intern/diff/D19609558)

[ghstack-poisoned]
